qml/pages/FirstPage.qml

```diff
@@ -41,67 +62,91 @@ Page {
             }
         }
 
-        SilicaListView {
-            id: listView
-            model: []
-            anchors.fill: parent
-            header: PageHeader {
-                title: qsTr("Current Broadcasts")
+        onModelChanged: rowCount = Math.ceil(model.length / columnCount)
+
+        // Cell width and height depend on columnCount, which depends on orientation
+        cellWidth: page.width / columnCount
+        cellHeight: page.width / columnCount / 16.0 * 9.0 + Theme.fontSizeMedium + Theme.paddingMedium
+
+        // Always, always remember to include the header itemi height...
+        contentWidth: width
+        contentHeight: gridView.headerItem.height + rowCount * cellHeight + offsetY
+
+        delegate: ListItem {
+            id: listItem
+            width: page.width / columnCount
+            property int baseHeight: gridView.cellHeight
+            height: baseHeight + (index === curreIndex ? offsetY : 0.0)
```

Comment:
Typo curreIndex should be gridView.currentIndex

Reply:
It actually should be global currIndex, as using gridView.currentView causes jerkiness when long-pressing the first element in portrait mode. But yes, it's a typo.